### ISSUE TITLE
state-advancer throws "Cannot mix BigInt and other types"

### ISSUE THREAD

**Issue by kloimhardt** [mentat-collective/emmy#133]
The following example (taken from the SICM book) throws the error "TypeError: Cannot mix BigInt and other types, use explicit conversions"

```
(defn harmonic-state-derivative [m k]
   (Lagrangian->state-derivative (L-harmonic m k)))

(def up01 (up 1.0 (up 1.0 2.0) (up 3.0 4.0)))

(def harmonic-state-advancer
  (state-advancer harmonic-state-derivative 2.0 1.0))

(harmonic-state-advancer up01, 10.0, 1.0e-12)
```

To reproduce, clone the [Emmy as ES6 Module](https://github.com/kloimhardt/emmy-module) repository, run `shadow-cljs watch emmy-esm` and open  `http://localhost:9000/barebones.html`

**Comment by sritchie** [comment]
Thank you! Would you mind pasting the error too?

**Comment by kloimhardt** [comment]
When I open [barebones.html](https://github.com/kloimhardt/emmy-module/blob/main/barebones.html) in the browser, the error 

`TypeError: Cannot mix BigInt and other types, use explicit conversions`

is shown on the screen.

**Comment by sritchie** [comment]
Ah, that pesky one... okay thank you, I will track it down!!

**Comment by sritchie** [comment]
Fixed by #134 and patched over at that repo by https://github.com/kloimhardt/emmy-module/pull/1.
